openreview/tools.py

```diff
@@ -16,37 +16,61 @@ def get_profile(client, value):
-			email_group = openreview.Group(id = email, signatures = [super_user_id], signatories = [email], readers = [email], writers = [super_user_id], members = [tilde_id])
-			profile_content = {
-		        'emails': [email],
-		        'preferred_email': email,
-		        'names': [
-		            {
-		                'first': first,
-		                'middle': middle,
-		                'last': last,
-		                'username': tilde_id
-		            }
-		        ]
-		    }
-			client.post_group(tilde_group)
-			client.post_group(email_group)
-			profile = client.post_profile(tilde_id, profile_content)
-
-			return profile
-
-		else:
-			raise openreview.OpenReviewException('There is already a profile with this first: {0}, middle: {1}, last name: {2}'.format(first, middle, last))
-	else:
-		raise openreview.OpenReviewException('There is already a profile with this email: {0}'.format(email))
+    profile = get_profile(client, email)
 
+    if not profile:
+        response = client.get_tildeusername(first, last, middle)
+        tilde_id = response['username'].encode('utf-8')
 
+        if tilde_id.endswith(last + '1'):
+
+            tilde_group = openreview.Group(id = tilde_id, signatures = [super_user_id], signatories = [tilde_id], readers = [tilde_id], writers = [super_user_id], members = [email])
+            email_group = openreview.Group(id = email, signatures = [super_user_id], signatories = [email], readers = [email], writers = [super_user_id], members = [tilde_id])
+            profile_content = {
+                'emails': [email],
+                'preferred_email': email,
+                'names': [
+                    {
+                        'first': first,
+                        'middle': middle,
+                        'last': last,
+                        'username': tilde_id
+                    }
+                ]
+            }
+            client.post_group(tilde_group)
+            client.post_group(email_group)
+            profile = client.post_profile(tilde_id, profile_content)
+
+            return profile
+
+        else:
+            raise openreview.OpenReviewException('There is already a profile with this first: {0}, middle: {1}, last name: {2}'.format(first, middle, last))
+    else:
+        raise openreview.OpenReviewException('There is already a profile with this email: {0}'.format(email))
+
+
+def build_groups(conference_group_id, default_params=None):
+    '''
+
+    Given a group ID, returns a list of empty groups that correspond to the given group's subpaths
+    (e.g. Test.com, Test.com/TestConference, Test.com/TestConference/2018)
+
+    '''
+
+    path_components = conference_group_id.split('/')
+    paths = ['/'.join(path_components[0:index+1]) for index, path in enumerate(path_components)]
+
+    if not default_params:
+        default_params = {
```

Comment:
is it useful to create an empty group?

May be we can send a user id as a parameter to set the signature and signatory. 

Are you using this function? where?

Reply:
Good point. I'm not sure if this is the best way to do it.

Right now, it works like this: when the "post_group" function gets called by the client, if there is an empty signature or writer field, the client fills it with the username that it was initialized with. (I remember that we discussed whether or not this was a good idea - I like it, but we can change this if you think it's not a good design).

That means that the empty fields are useful, since it's essentially saying that the writers and signatures will be set to whoever posts the group, and that there are no signatories or members by default. 

I have used this function once so far, to set up the groups for the NAACL preprint server. I call this function with the super user in a very simply jupyter notebook script like this:

```
import openreview
client = openreview.Client()
naacl_groups = openreview.tools.build_groups('naacl.org/NAACL/2018/Preprint')
for n in naacl_groups:
    client.post_group(n)
```

I haven't committed this code because I would like to design the python library (and the "builder" module) in a way that encourages interactivity.

Are these good design decisions? let me know if you disagree, I don't want to go down a bad path